File: php/src/Types/Heartbeat.php
<?php

declare(strict_types=1);

namespace K256\Types;

/**
 * Connection heartbeat with stats.
 */
readonly class Heartbeat
{
    public function __construct(
        /** Unix timestamp in milliseconds */
        public int $timestampMs,
        /** Connection uptime in seconds */
        public int $uptimeSeconds,
        /** Total messages received */
        public int $messagesReceived,
        /** Total messages sent */
        public int $messagesSent,
        /** Number of active subscriptions */
        public int $subscriptions,
    ) {}
}

File: rust/src/types/heartbeat.rs
//! Heartbeat types.

use serde::{Deserialize, Serialize};

/// Connection heartbeat with stats.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub struct Heartbeat {
    /// Unix timestamp in milliseconds
    pub timestamp_ms: u64,
    /// Connection uptime in seconds
    pub uptime_seconds: u64,
    /// Total messages received
    pub messages_received: u64,
    /// Total messages sent
    pub messages_sent: u64,
    /// Number of active subscriptions
    pub subscriptions: u32,
}
